pkg/sql/pgwire/testdata/auth/password_change

```diff
@@ -46,3 +48,34 @@ ok
 connect user=userpw password=pass2
 ----
 ERROR: password authentication failed for user userpw
+
+subtest end
+
+subtest root_pw
+
+# By default root cannot log in with a password.
+connect user=root sslmode=require sslcert= sslkey=
+----
+ERROR: password authentication failed for user root
+
+connect_unix user=root
+----
+ERROR: password authentication failed for user root
+
+
+# However if we give them a password, they can log in with password.
+sql
+ALTER USER root WITH PASSWORD 'secureabc'
+----
+ok
+
+# Then they can log ini.
+connect user=root password=secureabc sslmode=require sslcert= sslkey=
+----
+ok defaultdb
+
+connect_unix user=root password=secureabc
```

Comment:
Should we also test here connecting as root with a valid cert but incorrect password? That should work too right?